Initium-Odp/src/com/universeprojects/miniup/server/ODPDBAccess.java

```diff
@@ -4754,8 +4754,55 @@ else if (currentLocationKey.getId()==pathLocation2Key.getId())
 
 		if (destination.getProperty("ownerKey")!=null)
 		{
-			if (isInParty)
+			if (isInParty) {
+				List<CachedEntity> partyMembers = getParty(db, character);
+				List<CachedEntity> partyGroups = new ArrayList<CachedEntity>();
+				List<CachedEntity> partyUsers = new ArrayList<CachedEntity>();
+				for(CachedEntity partyMember: partyMembers) {
+					//Removes those who have just applied
+					String groupStatus = (String)character.getProperty("groupStatus");
+					if(("Member".equals(groupStatus)==false && "Admin".equals(groupStatus)==false)) {
+						partyGroups.add(new CachedEntity());
+					} else {
+						partyGroups.add(partyMember.getProperty("groupKey"));
+					}
+					partyUsers.add(partyMember.getProperty("userKey"));
+				}
+				Key ownerKey = (Key)destination.getProperty("ownerKey");
+				if(ownerKey.getKind().equals("Group")){
+					//iterates through the list of groups that the party members belong to.
+					//if there are any differences, set the flag to false
+					boolean allInSameGroup = true;
+					for(int i = 0; i < partyGroup.size() - 1; i++) {
+						if(!partyGroups.get(i).equals(partyGroups.get(i+1))){
+							allInSameGroup = false;
```

Comment:
I was considering that, but with the max party size being 4, I got a little lazy.  I can definitely look into doing that.